circleci: failed tests (13213): TestIntentResolution

The following test appears to have failed:

[#13213](https://circleci.com/gh/cockroachdb/cockroach/13213):

```
1      storage/replica.go:1776
1      server/node.go:558
I0222 21:08:02.485667 411 stopper.go:273  draining; tasks left:
1      storage/replica.go:1776
I0222 21:08:02.488108 411 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- FAIL: TestIntentResolution (1.57s)
	intent_test.go:137: 2: unexpected non-local intents, expected ["s"-"w" "s"-"x" "t"-"u" s]: ["s"-"w" "s"-"w" "s"-"x" "s"-"x" "t"-"u" "t"-"u" s s]
=== RUN   TestBootstrapCluster
I0222 21:08:02.502659 411 raft/raft.go:459  [group 1] 1 became follower at term 5
I0222 21:08:02.502905 411 raft/raft.go:235  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I0222 21:08:02.503036 411 raft/raft.go:539  [group 1] 1 is starting a new election at term 5
I0222 21:08:02.503153 411 raft/raft.go:472  [group 1] 1 became candidate at term 6
I0222 21:08:02.503274 411 raft/raft.go:521  [group 1] 1 received vote from 1 at term 6
I0222 21:08:02.503406 411 raft/raft.go:500  [group 1] 1 became leader at term 6
I0222 21:08:02.505747 411 storage/replica_command.go:1259  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1456175283.504s
I0222 21:08:02.511297 411 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--
I0222 21:08:20.152303 411 kv/range_cache.go:228  evict cached descriptor: key=/Meta1/System/"tsd!cr.node.exec.error-10m\x00\x01\x89\xf8\x06,\r1" desc=<nil>
I0222 21:08:20.152636 411 client/db.go:485  failed batch: failed to send RPC: too many errors encountered (1 of 1 total): server/node.go:559: node 1 stopped
W0222 21:08:20.152891 411 ts/db.go:96  error writing time series data: failed to send RPC: too many errors encountered (1 of 1 total): server/node.go:559: node 1 stopped
I0222 21:08:20.154588 411 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestMetricsRecording (2.78s)
FAIL
FAIL	github.com/cockroachdb/cockroach/server	22.462s
=== RUN   TestNodeStatusMonitor
--- PASS: TestNodeStatusMonitor (0.02s)
=== RUN   TestNodeStatusRecorder
--- PASS: TestNodeStatusRecorder (0.09s)
=== RUN   TestRuntimeStatRecorder
I0222 21:08:12.017883 438 server/status/runtime.go:137  runtime stats: 4 goroutines, 3.37MiB active, 50000000.00cgo/sec, 806020.00/2297180.00 %(u/s)time, 63744.28 %gc (3x)
--- PASS: TestRuntimeStatRecorder (0.00s)
=== RUN   TestNodeEventFeed
--- PASS: TestNodeEventFeed (0.03s)
=== RUN   TestServerNodeEventFeed

```
Please assign, take a look and update the issue accordingly.

--- a/server/intent_test.go
+++ b/server/intent_test.go
@@ -32,7 +32,6 @@ import (
 
 func TestIntentResolution(t *testing.T) {
 	defer leaktest.AfterTest(t)
-	t.Skip("https://github.com/cockroachdb/cockroach/issues/4579")
 
 	testCases := []struct {
 		keys   []string
@@ -69,9 +68,9 @@ func TestIntentResolution(t *testing.T) {
 	}
 
 	splitKey := []byte("s")
+	results := map[string]struct{}{}
 	defer func() { storage.TestingCommandFilter = nil }()
 	for i, tc := range testCases {
-		var result []string
 		var mu sync.Mutex
 		closer := make(chan struct{}, 2)
 		storage.TestingCommandFilter = func(_ roachpb.StoreID, args roachpb.Request, _ roachpb.Header) error {
@@ -79,18 +78,16 @@ func TestIntentResolution(t *testing.T) {
 			defer mu.Unlock()
 			header := args.Header()
 			// Ignore anything outside of the intent key range of "a" - "z"
-			// TODO: Implement "ContainsKey()" for Span, currently only
-			// implemented for RSpan
 			if header.Key.Compare(roachpb.Key("a")) < 0 || header.Key.Compare(roachpb.Key("z")) > 0 {
 				return nil
 			}
 			switch args.(type) {
 			case *roachpb.ResolveIntentRequest:
-				result = append(result, string(header.Key))
+				results[string(header.Key)] = struct{}{}
 			case *roachpb.ResolveIntentRangeRequest:
-				result = append(result, fmt.Sprintf("%s-%s", header.Key, header.EndKey))
+				results[fmt.Sprintf("%s-%s", header.Key, header.EndKey)] = struct{}{}
 			}
-			if len(result) == len(tc.exp) {
+			if len(results) == len(tc.exp) {
 				closer <- struct{}{}
 			}
 			return nil
@@ -129,14 +126,16 @@ func TestIntentResolution(t *testing.T) {
 			<-closer // wait for async intents
 		}()
 		// Verification. Note that this runs after the system has stopped, so that
-		// everything asynchronous has already happened (while tearing down, intent
-		// resolution still takes place, synchronously).
+		// everything asynchronous has already happened.
 		expResult := tc.exp
 		sort.Strings(expResult)
-		sort.Strings(result)
-		if !reflect.DeepEqual(result, expResult) {
-			t.Fatalf("%d: unexpected non-local intents, expected %s: %s", i, expResult, result)
+		var actResult []string
+		for k := range results {
+			actResult = append(actResult, k)
 		}
-
+		sort.Strings(actResult)
+		if !reflect.DeepEqual(expResult, expResult) {
+			t.Fatalf("%d: unexpected non-local intents, expected %s: %s", i, expResult, expResult)
+		}
 	}
 }

